File: Backend/SpringBoot/HostelManagement/src/main/java/com/project/hostelmanagement/entities/Room.java
package com.project.hostelmanagement.entities;

import java.util.List;

import org.springframework.stereotype.Service;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
	
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Entity
@Table(name = "room")
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Room_ID")
    private Integer roomId;

    @Column(name = "RoomNo",nullable = false)
    private Integer roomno;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RoomStatus status;
    
    @Column(name = "RoomType",nullable = false)
    private String roomtype;

    @Column(nullable = false)
    private Integer capacity; 

    @Column(nullable = false)
    private Float price; 

   
    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<RoomAllocation> roomAllocations; 

    
}

File: ASP.Net/Project/Models/Room.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Project.Models
{
    public partial class Room
    {
        public Room()
        {
            Roomallocations = new HashSet<Roomallocation>();
        }

        public int RoomId { get; set; }
        public int Capacity { get; set; }
        public float Price { get; set; }
        public int RoomNo { get; set; }
        public string RoomType { get; set; } = null!;
        public string Status { get; set; } = null!;

        [JsonIgnore]
        public virtual ICollection<Roomallocation> Roomallocations { get; set; }
    }
}
